src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int currentCapacity = 10;
+    private int size = 0;
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        grow();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        grow();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheckForGet(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeCheckForGet(index);
```

Comment:
The method name is confusing. on set and remove operations you use **get** check
rename to something like validateIndexExistance

Reply:
not fixed...